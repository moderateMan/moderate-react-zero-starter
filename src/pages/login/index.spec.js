// import dependencies
import React from 'react'

// import API mocking utilities from Mock Service Worker
import { rest } from 'msw'
import { setupServer } from 'msw/node'

// import react-testing methods
import { render, fireEvent, waitFor, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router'
import Login from '.'
describe('test Login page', () => {
  const server = setupServer(
    rest.post('/api/users/register', (req, res, ctx) => {
      return res(ctx.json({ greeting: 'hello there' }))
    })
  )

  beforeAll(() => server.listen())
  afterEach(() => server.resetHandlers())
  afterAll(() => server.close())

  test('handles server ok1', async () => {
    render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    )
    fireEvent.click(screen.getByRole('register'))
    let domInfo
    await waitFor(() => (domInfo = screen.getByRole('infoOk')))

    expect(domInfo).toHaveTextContent('登陆成功')
  })
  test('handles server ok', async () => {
    await waitFor(() => {
      server.use(
        rest.post('/api/users/register', (req, res, ctx) => {
          return res(ctx.status(500))
        })
      )
    })

    render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    )
    setTimeout(async () => {
      await waitFor(() => fireEvent.click(screen.getByRole('register')))

      let domInfo
      await waitFor(() => (domInfo = screen.getByRole('infoNologin')))

      expect(domInfo).toHaveTextContent('未登陆')
    }, 0)
  })

  test('renders correctly', async () => {
    const tree = render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
