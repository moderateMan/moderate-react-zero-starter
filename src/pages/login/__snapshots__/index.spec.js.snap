// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Login page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="ant-layout layout"
      >
        <div
          class="ant-card ant-card-bordered login_table"
        >
          <div
            class="ant-card-body"
          >
            <form
              class="ant-form ant-form-horizontal"
            >
              <div
                class="ant-row ant-form-item"
                style="width: 100%; margin: 10px;"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="name"
                    title="姓名"
                  >
                    姓名
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input ant-input-lg"
                        id="name"
                        maxlength="30"
                        placeholder="请输入"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-form-item"
                style="width: 100%; margin: 10px;"
              >
                <div
                  class="ant-col ant-form-item-label"
                >
                  <label
                    class="ant-form-item-required"
                    for="password"
                    title="密码"
                  >
                    密码
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        class="ant-input"
                        id="password"
                        placeholder="请输入"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </form>
            <div
              role="infoNologin"
            >
              未登陆
            </div>
            <div
              class="ant-row ant-row-center"
              style="margin-top: 20px;"
            >
              <button
                class="ant-btn ant-btn-default"
                style="margin-right: 20px;"
                type="button"
              >
                <span>
                  登 陆
                </span>
              </button>
              <button
                class="ant-btn ant-btn-default"
                role="register"
                type="button"
              >
                <span>
                  注 册
                </span>
              </button>
              <span
                class=""
              >
                <div
                  class="ant-upload ant-upload-select ant-upload-select-text"
                >
                  <span
                    class="ant-upload"
                    role="button"
                    tabindex="0"
                  >
                    <input
                      accept=""
                      style="display: none;"
                      type="file"
                    />
                    <button
                      class="ant-btn ant-btn-default"
                      type="button"
                    >
                      <span
                        aria-label="upload"
                        class="anticon anticon-upload"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="upload"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                          />
                        </svg>
                      </span>
                      <span>
                        Click to Upload
                      </span>
                    </button>
                  </span>
                </div>
                <div
                  class="ant-upload-list ant-upload-list-text"
                />
              </span>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="ant-layout layout"
    >
      <div
        class="ant-card ant-card-bordered login_table"
      >
        <div
          class="ant-card-body"
        >
          <form
            class="ant-form ant-form-horizontal"
          >
            <div
              class="ant-row ant-form-item"
              style="width: 100%; margin: 10px;"
            >
              <div
                class="ant-col ant-form-item-label"
              >
                <label
                  class="ant-form-item-required"
                  for="name"
                  title="姓名"
                >
                  姓名
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input ant-input-lg"
                      id="name"
                      maxlength="30"
                      placeholder="请输入"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-row ant-form-item"
              style="width: 100%; margin: 10px;"
            >
              <div
                class="ant-col ant-form-item-label"
              >
                <label
                  class="ant-form-item-required"
                  for="password"
                  title="密码"
                >
                  密码
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input"
                      id="password"
                      placeholder="请输入"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </form>
          <div
            role="infoNologin"
          >
            未登陆
          </div>
          <div
            class="ant-row ant-row-center"
            style="margin-top: 20px;"
          >
            <button
              class="ant-btn ant-btn-default"
              style="margin-right: 20px;"
              type="button"
            >
              <span>
                登 陆
              </span>
            </button>
            <button
              class="ant-btn ant-btn-default"
              role="register"
              type="button"
            >
              <span>
                注 册
              </span>
            </button>
            <span
              class=""
            >
              <div
                class="ant-upload ant-upload-select ant-upload-select-text"
              >
                <span
                  class="ant-upload"
                  role="button"
                  tabindex="0"
                >
                  <input
                    accept=""
                    style="display: none;"
                    type="file"
                  />
                  <button
                    class="ant-btn ant-btn-default"
                    type="button"
                  >
                    <span
                      aria-label="upload"
                      class="anticon anticon-upload"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="upload"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M400 317.7h73.9V656c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V317.7H624c6.7 0 10.4-7.7 6.3-12.9L518.3 163a8 8 0 00-12.6 0l-112 141.7c-4.1 5.3-.4 13 6.3 13zM878 626h-60c-4.4 0-8 3.6-8 8v154H214V634c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v198c0 17.7 14.3 32 32 32h684c17.7 0 32-14.3 32-32V634c0-4.4-3.6-8-8-8z"
                        />
                      </svg>
                    </span>
                    <span>
                      Click to Upload
                    </span>
                  </button>
                </span>
              </div>
              <div
                class="ant-upload-list ant-upload-list-text"
              />
            </span>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
